### PR TITLE
add python3 support

### PR DESCRIPTION
**Pull request by fsaintma**

### CODE REVIEW

File: iotlabaggregator/connections.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def handle_close(self):
 
     def handle_read(self):
         """ Append read bytes to buffer and run data handler. """
-        self.data_buff += self.recv(8192)
+        self.data_buff += (self.recv(8192)).decode('utf-8-', 'replace')
```
Comment (rienafairefr) [inline iotlabaggregator/connections.py]: ça devrait etre 'utf-8' et pas 'utf-8-' ??

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+    color:      {[testenv:color]commands}
+    coverage:   {[testenv:coverage]commands}
+
+[testenv:tests]
+commands=
+    py.test --cov=iotlabaggregator --cov-report=term --cov-report=xml --cov-report=html
+
+[testenv:color]
+commands=
+    pip install -e .[color_serial]
```
Comment (aabadie) [inline tox.ini]: is `-e` option really required ? (it creates symlinks)

**Comment by rienafairefr** [comment]
+ change the version number with a breaking change:
__version__ = '1.5.0'   => 2.xx
install_requires=['iotlabcli>=2.1.0'] => requires iotlabcli version 3

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,31 @@
 [tox]
-envlist = copying,py27{,-color}
+envlist = copying,{py34,py35,py36,py37}-{lint,tests,color,cli}
+
+[testenv]
+whitelist_externals =
+    cli:   {[testenv:cli]whitelist_externals
+deps=
+    git+git://github.com/iot-lab/cli-tools.git#egg=iotlabcli
```
Comment (aabadie) [inline tox.ini]: Why not just `iotlabcli` ?

Comment (rienafairefr) [reply tox.ini]: probably because `iotlabcli` on pypi is not the right version ?

Comment (aabadie) [reply tox.ini]: I don't know, it was already like this before.

File: iotlabaggregator/connections.py (hunk 1 above)
Comment (fsaintma) [reply iotlabaggregator/connections.py]: yes I will fix that and it's not utf-8 for sniffer

File: tox.ini (hunk 3 above)
Comment (fsaintma) [reply tox.ini]: @aabadie what is the question ? I don't understand :)

File: tox.ini (hunk 2 above)
Comment (fsaintma) [reply tox.ini]: > * change the version number with a breaking change:
>   **version** = '1.5.0'   => 2.xx
>   install_requires=['iotlabcli>=2.1.0'] => requires iotlabcli version 3

@rienafairefr For me we don't need a iotlabcli version 3. We just need a version of iotlabcli which supports python version 3. So 2.2.1 version will be a good option ?

Comment (rienafairefr) [reply tox.ini]: you're right, iotlabcli does not need to be version 3 :-)

And the version of aggregation-tools that supports only python 3 should be a breaking version change

File: tox.ini (hunk 3 above)
Comment (rienafairefr) [reply tox.ini]:
```
deps=
    iotlabcli
```
might be simpler, why use the github master version ?

**Comment by fsaintma** [comment]
@aabadie, @rienafairefr I have tested this code on the DEV Grenoble frontend SSH (virtualenv with python3.5) and it works with sniffer and serial aggregator. It seems to be good ?

**Pull request merged**